src/libraries/System.Formats.Asn1/src/System/Formats/Asn1/AsnWriter.Oid.cs

```diff
@@ -140,9 +141,14 @@ private static BigInteger ParseSubIdentifier(ref ReadOnlySpan<char> oidValue)
                 throw new ArgumentException(SR.Argument_InvalidOidValue, nameof(oidValue));
             }
 
-            // The following code is equivalent to
-            // BigInteger.TryParse(temp, NumberStyles.None, CultureInfo.InvariantCulture, out value)
-            // TODO: Split this for netstandard vs netcoreapp for span-perf?.
+#if NETCOREAPP2_1_OR_GREATER
+            if ((endIndex > 1 && oidValue[0] == '0') ||
+                !BigInteger.TryParse(oidValue.Slice(0, endIndex), NumberStyles.None, CultureInfo.InvariantCulture, out BigInteger value))
+            {
+                // T-REC X.680-201508 sec 12.26
+                throw new ArgumentException(SR.Argument_InvalidOidValue, nameof(oidValue));
+            }
+#else
```

Comment:
The suggested equivalent code in the comment is not actually spec compliant because `BigInteger.TryParse` will allow leading zeros on the input. However T-REC X.680-201508 spec does not allow that for the subidentifiers; for example `1.01` is invalid.

This split also causes some minor regressions for very small subidentifiers such as `1.1.1.1.1.1` because of the BigInteger parsing expecting large integers. The original loop performs suprisingly well here so I'm down to just revert this.

Reply:
Yeah, I think Oid_MostArcs shows that this is a net bad; so the change should probably be just to delete the comment that suggests splitting it :smile:

Reply:
Reverted in https://github.com/dotnet/runtime/pull/72055/commits/29c76109921d5be8c112b807cc77568524d3e421